test/runtests.jl

```diff
@@ -111,5 +111,53 @@ using Test
             @test call1 != call3
             @test call2 != call4
         end
+
+        @testset "splatting" begin
+            f(a, b; kw1=1, kw2=2) = a + b + kw1 + kw2
+
+            one = 1
+            a = [1, 2]
+            kw = (kw1=1, kw2=2)
+
+            @testset "basic splatting" begin
+                @test f(a...) == @cached_call f(a...)
+                @test f(a...) == @cached_call f([1, 2]...)
+                @test f(a...) == @cached_call f([one, 2]...)
+                @test f(a...; kw...) == @cached_call f(a...; kw...)
+                @test f(a...; kw...) == @cached_call f(a...; (kw1=1, kw2=2)...)
+                @test f(a...; kw...) == @cached_call f(a...; (kw1=one, kw2=2)...)
+
+                @test @hash_call(f(a...)) isa UInt
+                @test @hash_call(f([1, 2]...)) isa UInt
+                @test @hash_call(f([one, 2]...)) isa UInt
+                @test @hash_call(f(a...; kw...)) isa UInt
+                @test @hash_call(f(a...; (kw1=1, kw2=2)...)) isa UInt
+                @test @hash_call(f(a...; (kw1=one, kw2=2)...)) isa UInt
+            end
+
+            @testset "different containers" begin
+                call1 = @cached_call f(1, 2)
+                call2 = @cached_call f([1, 2]...)
+                call3 = @cached_call f((1, 2)...)
+                @test call1 == call2 == call3
+
+                @test @hash_call(f([1, 2]...)) isa UInt
+                @test @hash_call(f((1, 2)...)) isa UInt
+            end
+
+            @testset "make sure we hash splatted values (not variable names)" begin
```

Comment:
> make sure we hash splatted values (not variable names)

What does this comment mean? 

This is my first look at CachedCalls - gave it a quick try locally.

Reply:
We want to make sure that the hashes are computed based on the values of the arguments, not their names. I.e. for the normal arguments, we want:
```
a = 1
b = 1
h1 = @hash_call f(a)
h2 = @hash_call f(b)
@test h1 == h2

a = 1000
h3 = @hash_call f(a)
h1 != h3
```
This might seem paranoid if one is used to normal functions, but macros have access to the name of the variable used as the argument (`a`). If we made a mistake in the macro body we could be hashing any call `f(a)` to the same hash regardless of what the value of `a` is.

Reply:
Ah I see. I was a bit confused because we have keyword arguments floating around here and I would expect this to account for keyword argument names.